### PR TITLE
#19070 almost all template properties working

### PR DESCRIPTION
**Pull request by dsilvam**
Implements the `template` field for the GraphQL Page API. 
Returns the same fields returned by the REST API. 
Also Postman tests were included.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.DotContentletTransformer;
+import com.dotmarketing.portlets.contentlet.transform.DotTransformerBuilder;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:10]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.contentlet.transform.DotTransformerBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.DotContentletTransformer;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.contentlet.transform.DotContentletTransformer'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * to find a specific version of the page: languageId and pageMode.
+ *
+ * The returned page includes extra properties set by a page transformer.
+ *
+ */
+public class TemplateDataFetcher implements DataFetcher<Map<Object, Object>> {
+    @Override
+    public Map<Object, Object> get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            final User user = ((DotGraphQLContext) environment.getContext()).getUser();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:38]: Codacy found an issue: [Avoid unused local variables such as 'user'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+    private Template getTemplate(final String templateId,
+            final PageMode mode) throws DotDataException {
+        try {
+            final User systemUser = APILocator.getUserAPI().getSystemUser();
+
+            return mode.showLive ?
+                    (com.dotmarketing.portlets.templates.model.Template)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:66]: Codacy found an issue: [Unnecessary use of fully qualified name 'com.dotmarketing.portlets.templates.model.Template' due to existing import 'com.dotmarketing.portlets.templates.model.Template'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:38]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                    != null ? environment.getArgument("pageMode") : PageMode.LIVE.name();
+
+            final PageMode mode = PageMode.get(pageModeAsString);
+
+            final String templateId = contentlet.getStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD);
+
+            final Template template = getTemplate(templateId, mode);
+
+            return asMap(template);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:54]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * The returned page includes extra properties set by a page transformer.
+ *
+ */
+public class TemplateDataFetcher implements DataFetcher<Map<Object, Object>> {
+    @Override
+    public Map<Object, Object> get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            final User user = ((DotGraphQLContext) environment.getContext()).getUser();
+            final HttpServletRequest request = ((DotGraphQLContext) environment.getContext())
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:39]: Codacy found an issue: [Avoid unused local variables such as 'request'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:39]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:38]: Codacy found an issue: [Avoid unused local variables such as 'request'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    }
+
+    private Template getTemplate(final String templateId,
+            final PageMode mode) throws DotDataException {
+        try {
+            final User systemUser = APILocator.getUserAPI().getSystemUser();
+
+            return mode.showLive ?
+                    (com.dotmarketing.portlets.templates.model.Template)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:65]: Codacy found an issue: [Unnecessary use of fully qualified name 'com.dotmarketing.portlets.templates.model.Template' due to existing import 'com.dotmarketing.portlets.templates.model.Template'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:38]: Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+                    != null ? environment.getArgument("pageMode") : PageMode.LIVE.name();
+
+            final PageMode mode = PageMode.get(pageModeAsString);
+
+            final String templateId = contentlet.getStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD);
+
+            final Template template = getTemplate(templateId, mode);
+
+            return asMap(template, user);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:53]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

**Review by freddyucv — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/TemplateDataFetcher.java:45]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5999310)

**Pull request merged**